README.md

```diff
@@ -100,6 +100,38 @@ resource "ironic_allocation_v1" "openshift-master-allocation" {
 }
 ```
 
+## Deployment
+
+A deployment will provision a baremetal node, using the information
+given in the resource.  The `count` metaparameter can be used to deploy
+multiple nodes. Terraform will drive the [Ironic state
+machine](https://docs.openstack.org/ironic/latest/contributor/states.html)
+to bring the node to an `active` state.  The separation of deployment
+from the baremetal node defintion allows a user to manage their hardware
+outside of the provider if desired. Destruction of a deployment brings
+the baremetal node back to the `available` state.
+
+Users may specify a `node_uuid` directly, or make use of the allocation
+resource to dynamically pick a node.
+
+
+```terraform
+resource "ironic_deployment" "masters"  {
```

Comment:
So how does it affect in terms of defining terraform resources? We will need to define resource "ironic_node_v1" "openshift-master"  + resource "ironic_deployment" "masters" ?
For count you could set: count = "${length(keys(var.master_nodes))}"

Reply:
Yes, for now that's how we'd do it.

Eventually we could use existing hardware in Ironic and not need to specify a node at all, we'd ask Ironic for an [allocation](https://specs.openstack.org/openstack/ironic-specs/specs/not-implemented/allocation-api.html), and then use a deployment.